src/App.tsx

```diff
@@ -1,222 +1,225 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {
+  getTodos, patchTodos, deleteTodo, postTodos,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/Error/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [loadingTodosId, setLoadingTodosId] = useState<number[]>([]);
+  const [currentFilter, setCurrentFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+
+  const loadTodos = async () => {
+    try {
+      const TodosFromServer = await getTodos(user?.id || 0);
+
+      setTodos(TodosFromServer);
+    } catch {
+      setError('Error to load user todos');
+    } finally {
+      setError('');
+    }
+  };
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (!user) {
+      return;
+    }
+
+    loadTodos();
   }, []);
 
+  const addTodo = async (newTodo: Todo) => {
+    try {
+      setIsAdding(true);
+      setTodos(currentTodos => [...currentTodos, newTodo]);
+    } catch {
+      setError('Can`t add todo');
+    } finally {
+      setError('');
+      setIsAdding(false);
+    }
```

Comment:
you should put this try catch block inside a async function, here you won't catch anything